salesforcecli/plugin-info
# Ew/get release notes

### What does this PR do?
Initial functionality of the contextual notifications plugin. 

- Command name is `info:releasenotes:display`
  - Create an alias called `whatsnew`
- Add ability to suppress the notes with an env var
  - Decided not to use config because of the differences in `sf` and `sfdx`. Main goal for this was to suppress release notes during CI, which folks will use env vars anyway.
- Gets config from cli's `pjson.oclif` config 
- Ability to use dist-tag helpers to look up release notes
  - Supports `stable`, `latest`, `stable-rc`, `latest-rc`, `rc`
  - Collect version from the npm dist-tag link
- Looks first for a readme based on the major version (`vX.md`). If that is not found, use README.md
  - If no version is passed, uses installed version


### Testing Instructions
This plugin will get configuration from the cli's pjson. You will need to `sfdx plugins:link .` this branch and then add the following to your installed `sfdx` pjson in the `oclif` section:
```
"info": {
  "releasenotes": {
    "distTagUrl": "https://registry.npmjs.org/-/package/sfdx-cli/dist-tags",
    "releaseNotesPath": "https://raw.githubusercontent.com/forcedotcom/cli/main/releasenotes/sfdx",
    "releaseNotesFilename": "README.md"
  }
}
```
Then try out some commands
- `sfdx info:releasenotes:display`
- Also try the alias: `sfdx whatsnew`
- `sfdx whatsnew -h`
- `sfdx whatsnew -v 50.0.0`
  - Note that this returns the release notes from [v50.md](https://github.com/forcedotcom/cli/blob/main/releasenotes/sfdx/v50.md)
- `sfdx whatsnew -v latest` (gets README.md)
- `sfdx whatsnew -v stable-rc` (gets README.md)
- `sfdx whatsnew -v rc` (gets README.md)
- `sfdx whatsnew` (uses installed version)

### What issues does this PR fix or reference?
[@W-9996068@](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-9996068)
